### ISSUE TITLE
Cargo audit findings

### ISSUE THREAD

**Issue by elsirion** [fedimint/fedimint#3010]
[`cargo audit` is complaining about the following dependencies](https://github.com/fedimint/fedimint/actions/runs/5940802339/job/16110189706?pr=3009):

* [atty v0.2.14](https://rustsec.org/advisories/RUSTSEC-2021-0145)
* [webpki v0.22.0](https://rustsec.org/advisories/RUSTSEC-2023-0052)
* [webpki v0.21.4](https://rustsec.org/advisories/RUSTSEC-2023-0052)
* [rustls v0.100.0](https://rustsec.org/advisories/RUSTSEC-2023-0053)

**Comment by elsirion** [comment]
None of these seem super critical to fix: either DoS (we have a lot of DoS potential vectors in our own code) and the `atty` exploit only works on windows. Let's still try to get rid of them, but we have the time to coordinate with e.g. the ongoing project to update the `rust-bitcoin` dependency (#2667) which will likely unblock newer `electrum-client` versions.

**Comment by elsirion** [comment]
Closing in favor of #3049 since most items here were fixed.
